### DISCUSSION TITLE
Custom server renderError

### DISCUSSION THREAD

**Discussion by Antignote** [vercel/next.js#63912] [Help]
### Summary

How can I accomplish to render the src/app/error.tsx using a custom server and app router in nextjs 14?

I try to use the [renderError](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/next.ts#L128) function on the returned router from calling `next()`.
The goal is that I want to reuse the same error page if an exception is occurred in my Express.js server.

```ts
const app = next({/* otpions */});
// ...
app.renderError(err, req, res, '/', {});
```

AFAIK i get stuck on await `this.ready?.promise` in the router since the `NextCustomServer.prepare` didn't setup this.ready promise in the `prepareImpl` function.

Any ideas? I can help to contribute to the Next.js project if the case would be that this isn't implemented atm.

**Comment by StNekroman** [comment]
That's is a bug in next.js which nobody wants to fix.
I hit this too.
If you create NextCustomServer -  you just cannot use methods like "renderError", because nobody prepared the render servier itself.
There was already one opened issue on this, but somebody from vercel team closed it with message "not enough people use custom server, and "renderError" is not public methed"
While according to typings - it's public.
